CRM/Case/Form/CaseView.php

```diff
@@ -55,6 +55,12 @@ public function preProcess() {
 
     $this->assign('contactID', $this->_contactID = (int) $this->get('cid'));
     $this->assign('caseID', $this->_caseID = (int) $this->get('id'));
+    $this->assign('primaryCaseContactID', civicrm_api3('Activity', 'get', [
+      'case_id' => $this->_caseID,
+      'activity_type_id' => 'Open Case',
+      'return' => ['assignee_contact_id', 'source_contact_id', 'target_contact_id'],
+      'sequential' => 1,
+    ])['values'][0]['target_contact_id'][0]);
```

Comment:
This gives a warning `Warning: Undefined array key 0 in CRM_Case_Form_CaseView->preProcess() (line 63` when there is no open case in the config. It's not a common configuration and not recommended but I've seen it out in the wild and we know from chat threads it exists. There is `CRM_Case_BAO_Case::getCaseClients`?

Reply:
@demeritcowboy that wouldn't tell me which of the n case clients has the activities of the case set to them right?

Reply:
Maybe we need a clearer description of what the user use-cases are and why the need for the concept of Primary. In the most general case I don't see why someone can't remove ANY of the clients as long as you can't remove the "last" one (when there's only one left/existing). If there's a finer requirement for "don't let me remove one where there's an activity where they are the target", then we should spell that out (and unit test 😈).

Reply:
@demeritcowboy don't mention that u word :-P, so basically Brian came to me with a problem that you couldn't remove the Additional Clients when you have multiple clients. 

I found that you could remove the 'primary' one i.e. the one with the activities attached to it by using the EditClient form (which should really be call re-assign case but anywho) What that does do is re-builds the case with essentially a new 'primary' case contact i.e. the one that is the target on the activities.  

So I was thinking we would need some difference between the other cases clients i.e. the ones that don't have any activities linked to them and the one that has the activities linked to them you know.

Reply:
The EditClient/Reassign isn't tied to activities, and there is no real concept of "primary". It depends on what url you followed to get to Manage Case. The cid in the url can be any of the client ids. If you come from one of their contact Case tabs then the reassign button will reassign that person, completely separate from what is on the activities.

So it sounds like the biz requirement is just to be able to remove any of the clients. So the only restriction/lookup would be that you shouldn't end up with a case with 0 clients, i.e. the "x" shouldn't be shown if there is only one client. And ideally the preprocess() for the delete form would bounce if you visit a crafted url directly and there is only one client.

Reply:
@demeritcowboy so there will be one client that is attached to various activities  based on this https://github.com/civicrm/civicrm-core/blob/master/CRM/Case/Form/Activity/OpenCase.php#L214 / https://github.com/civicrm/civicrm-core/blob/master/CRM/Case/Form/Activity/OpenCase.php#L294 so what happens if we decide to view one of the other clients and then go to remove the one that was originally linked to the activities should this form just move those activities to the currently viewed contact?

Reply:
I suppose that could be an optional checkbox option, but it would depend on the given install and what cases represent in that install and what multiple clients means in that install, and why they've chosen to remove that client, and whether it makes sense in that situation to "change history". In general it's perfectly fine to have activities on a case with any target.

Since I'm the one who usually overthinks things, it feels weird for me to type that maybe this PR doesn't need that much brainwave activity, although I appreciate it.

Everything that's been said so far suggests it really just needs a way to remove clients, but don't end up with 0 clients.

Reply:
I agree. I don't think we should automatically relink activities. This should be kept simple and straightforward.